src/OrleansBenchmarks/MapReduce/MapReduceBenchmark.cs

```diff
@@ -0,0 +1,175 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using BenchmarkDotNet.Attributes;
+using GrainInterfaces;
+using Orleans;
+using Orleans.Concurrency;
+using Orleans.Runtime;
+using Orleans.Runtime.Configuration;
+using Orleans.Runtime.Host;
+using Orleans.Serialization;
+
+namespace SerializationBenchmarks.MapReduce
+{
+    public class MapReduceBenchmark
+    {
+        private static SiloHost _siloHost;
+        private readonly int _intermediateStagesCount = 15;
+        private readonly int _pipelineParallelization = 4;
+        private readonly int _repeats = 50000;
+        private int _currentRepeat = 0;
+
+        [Setup]
+        public void BenchmarkSetup()
+        {
+            AppDomain hostDomain = AppDomain.CreateDomain("OrleansHost", null, new AppDomainSetup
+            {
+                AppDomainInitializer = InitSilo,
+                AppDomainInitializerArguments = new string[] { "Primary", "BenchServerConfiguration.xml" },
+            });
```

Comment:
Should we be using `TestCluster` instead to manage this for us? It will be simpler for when we need more than 1 silo (or maybe that will never be an option for these kinds of benchmarks?).
Also it will probably be easier if we migrate this to .net core, since once we update `TestCluster` to run silos without new AppDomains (as it's not supported), this will just work, otherwise this benchmark might require keeping track of started processes to run the silos.